index.bs

```diff
@@ -1300,6 +1302,10 @@ A [=view=] has an associated <dfn for="view">eye</dfn> which is an {{XREye}} des
 
 Note: Many HMDs will request that content render two [=views=], one for the left eye and one for the right, while most magic window devices will only request one [=view=], but applications should never assume a specific view configuration. For example: A magic window device may request two views if it is capable of stereo output, but may revert to requesting a single view for performance reasons if the stereo output mode is turned off. Similarly, HMDs may request more than two views to facilitate a wide field of view or displays of different pixel density.
 
+User-agents MAY choose to expose more than two [=views=] at a time, and content should be written to expect any nonzero number of elements in the [=views=] array, to support things like quad-view devices (where there are two extra low-resolution views with wider fields of view), or CAVE systems.
```

Comment:
Since almost all content will only work on `mono` or `stereo` devices, maybe a feature that lets the site opt into multiple views would be better.

Reply:
Yeah I was considering this! I'm totally open to that option as well, I wasn't sure which was better. I was a bit concerned because I wasn't sure if content could get this right for the scope of _all_ future secondary views. E.g. quad views give you two views for each eye, which is an additional assumption broken.

@toji, what do you think? I could instead write spec text for a `secondary-views` feature in this spec, and first-person observer views just become a spec concept (and not a feature) in the AR module. Such a feature would basically be asking authors to certify:

 - they expect the views array to change size
 - they can handle any number of views
 - they can handle multiple views with the same eye, with the stipulation that the first two will always be the primary L/R views if necessary

Reply:
I'm not sure about that, especially if we can get the frameworks like babylon.js or three.js to support multiple views.

Reply:
What if your experience uses multiview? Will that easily work if there are suddenly more displays?

Reply:
@asajeffrey the feedback from @thetuvix was that:

 - most applications rarely expect extra views
 - _specifically_ in the first-person observer case, if the application is not going to handle extra views then we should instead let the device APIs synthesize an FPO view over exposing an unused view

so there's a general feeling that content should signal support for extra views, and a stronger feeling that _at least_ FPO views should be specifically requested

Reply:
Our experience from HoloLens is that the easy part is getting an engine like Unity or Babylon to plop a scene camera at every view and produce some pixels.  However, in practice, apps scripted in those engines tend to assume stereo unless tested from that extra perspective, and so their own shaders will make broken assumptions (e.g. in instancing, assumptions about the view origin, etc.).

It is also valuable to know whether a given app/page has opted-in because then you can do a better fallback for the pages that don't:
* On HoloLens 2, if an app doesn't opt into the first-person observer view, we do the least-bad thing and distort the left or right eye image.  While this isn't ideal, it's better than rendering black or an otherwise broken image from an app that wasn't hardened correctly for that third view.
* On Starbreeze quad view headsets with wide-left and wide-right displays, if the app knows how to render 4 views, it's given one view that represents each display.  If the app doesn't opt-in, however, it's given 2 stereo views that are each larger and overlap multiple displays.  They don't cover the device's full FOV, but they cover much more than if the app rendered for the inner displays and then provided left the buffers black for the outer displays.

Reply:
Yeah, it seems like this should be per-view-configuration-style opt in then.